src/styles/blocks/days.scss

```diff
@@ -0,0 +1,42 @@
+.days {
+  padding: $padding;
+  display: flex;
+  flex-wrap: wrap;
+  gap: $gap;
+
+  @each $day, $num in $start-day {
+    &--start-day_#{$day} {
+      .days__day:first-child {
+        margin-left: ($days-size +$gap) * $num;
+      }
+    }
+  }
+
+  &__day {
+    @include square($days-size);
+    border: 1px solid #000;
+    background-color: #eee;
+    transition: $transition;
+
+    &:hover {
+      background-color: pink;
+      transform: translateY(-20px);
+      cursor: pointer;
+    }
+
+    @for $index from 1 through 31 {
+      &:nth-child(#{$index}):before {
+        content: "#{$index}";
+        @extend %number-properties;
+      }
+    }
+  }
```

Comment:
That's what I meant by Fix the loop.
```suggestion
  &__day {
    @include square($days-size);
    border: 1px solid #000;
    background-color: #eee;
    @extend %number-properties;
    transition: $transition;

    &:hover {
      background-color: pink;
      transform: translateY(-20px);
      cursor: pointer;
    }

    @for $index from 1 through 31 {
      &:nth-child(#{$index}):before {
        content: "#{$index}";
      }
    }
  }
```

Reply:
Oh fixed. I kept thinking :after falls outside the &_day element.